django_mongodb/query_utils.py

```diff
@@ -55,3 +55,17 @@ def wrapper(value):
 
     wrapper.__name__ = "safe_regex (%r)" % regex
     return wrapper
+
+
+def regex_match_expression(field, value, regex, *re_args, **re_kwargs):
```

Comment:
@Jibola I was not able to pass the re.compile regex to MongoDB using $regexMatch, I unfold the regex but the `u` flag is not working. With mongosh do work, but it does not with pymongo.  Maybe a pymongo bug?

Reply:
Could you provide a code example? Maybe there's a proper workaround help debug with.

Reply:
But what's the failing case?

I tried `lookup_article.count_documents({'$expr': {'$regexMatch': {'input': '$headline', 'regex': '^What¥', 'options': ''}}})` and it seemed to match a unicode character just fine -- no special option needed.

`re.U`: "In Python 3, Unicode characters are matched by default for str patterns. This flag is therefore redundant with no effect and is only kept for backward compatibility."

Reply:
You are right. So, we can ignore the u flag.

Reply:
And re.S / re.M aren't currently used, right? In that case, I would probably omit them in `regex_match` since it isn't a public API we expect users to call with their own options.

Incidentally, maybe you want to use just one of either `re.I` or `re.IGNORECASE` for consistency.

Reply:
Yeap, ignore is the only one that matters. I will only keep that.